packages/web-client/tests/acceptance/create-card-space-persistence-test.ts

```diff
@@ -0,0 +1,201 @@
+
+  hooks.beforeEach(async function () {
+    let layer2Service = this.owner.lookup('service:layer2-network')
+      .strategy as Layer2TestWeb3Strategy;
+    layer2Service.test__simulateAccountsChanged([layer2AccountAddress]);
+
+    layer2Service.authenticate();
+    layer2Service.test__simulateHubAuthentication('abc123--def456--ghi789');
+
+    workflowPersistenceService = this.owner.lookup(
+      'service:workflow-persistence'
+    );
+
+    workflowPersistenceService.clear();
+  });
+
+  test('Generates a flow uuid query parameter used as a persistence identifier', async function (this: Context, assert) {
+    await visit('/card-space');
+    await click('[data-test-workflow-button="create-space"]');
+
+    assert.equal(
+      // @ts-ignore (complains object is possibly null)
+      new URL('http://domain.test/' + currentURL()).searchParams.get('flow-id')
+        .length,
+      22
+    );
+  });
+
+  module('Restoring from a previously saved state', function () {
+    test('it restores an unfinished workflow', async function (this: Context, assert) {
+      let state = buildState({
+        meta: {
+          completedCardNames: ['LAYER2_CONNECT', 'CARD_SPACE_USERNAME'],
+        },
+      });
+
+      workflowPersistenceService.persistData('abc123', {
+        name: 'CARD_SPACE_CREATION',
+        state,
+      });
+
+      await visit('/card-space?flow=create-space&flow-id=abc123');
+
+      assert.dom(milestoneCompletedSel(0)).exists(); // L2 connect
+      assert.dom(milestoneCompletedSel(1)).exists(); // Username
+      assert.dom(milestoneCompletedSel(2)).doesNotExist();
+      assert.dom('[data-test-milestone="2"][data-test-postable="2"]').exists();
+      assert.dom('[data-test-card-space-details-start-button]').isNotDisabled();
+    });
+
+    test('it restores a finished workflow', async function (this: Context, assert) {
+      let state = buildState({
+        meta: {
+          completedCardNames: [
+            'LAYER2_CONNECT',
+            'CARD_SPACE_USERNAME',
+            'CARD_SPACE_DETAILS',
+            'CARD_SPACE_CONFIRM',
+          ],
+        },
+      });
+
+      workflowPersistenceService.persistData('abc123', {
+        name: 'CARD_SPACE_CREATION',
+        state,
+      });
+
+      await visit('/card-space?flow=create-space&flow-id=abc123');
+
+      assert.dom(milestoneCompletedSel(0)).exists(); // L2 connect
+      assert.dom(milestoneCompletedSel(1)).exists(); // Username
+      assert.dom(milestoneCompletedSel(2)).exists(); // Details
+      assert.dom(milestoneCompletedSel(3)).exists(); // Confirm
+      assert
+        .dom('[data-test-epilogue][data-test-postable="1"]')
+        .includesText(`Congrats, you have created your Card Space!`);
+    });
+
+    test('it restores a cancelled workflow', async function (this: Context, assert) {
+      const state = buildState({
+        meta: {
+          completedCardNames: ['LAYER2_CONNECT', 'CARD_SPACE_USERNAME'],
+          isCancelled: true,
```

Comment:
Typescript has [saved us](https://github.com/cardstack/cardstack/pull/2202/checks?check_run_id=3804262948#step:6:15) re the spelling 😆

Reply:
> Typescript has [saved us](https://github.com/cardstack/cardstack/pull/2202/checks?check_run_id=3804262948#step:6:15) re the spelling 😆

It tells me the opposite when I try to fix it 😩
<img width="1124" alt="ts" src="https://user-images.githubusercontent.com/16160806/136054698-7de148d4-7f67-46b2-90ed-d448bfca2df4.png">

Reply:
maybe you need to rebase on `main`? It looks like you branched off at 8816e01 which is just before #2197 was merged